### ISSUE TITLE
`pseudolog` transformer

### ISSUE THREAD

**Issue by ParadaCarleton** [feature-engine/feature_engine#707]
Request for an inverse-hyperbolic-sine, a.k.a. asinh or pseudolog, transformer. `x->arcsinh(x/2)` behaves like `ln(x)` for large values of `x`, but behaves like `x->x` for small values of `x`; this behavior is very useful for values that are almost lognormal, but take on both positive and negative values (e.g. net worth).

Ideally, this should provide location and scale parameters that can be either tuned or set to 0/1.

**Comment by solegalli** [comment]
Hi Carlos,

Thanks for the suggestion. I am not familiar with this transformation, so I don't undertand what you mean by location and scale parameters and tune to 0/1.

Do you have a resource with more details about this transformation that you could share? Like when is it used? who developed it, or whatever you have at hand? We would need that in any case to create the documentation.

Thank you!

**Comment by ParadaCarleton** [comment]
> Thanks for the suggestion. I am not familiar with this transformation, so I don't undertand what you mean by location and scale parameters and tune to 0/1.

You can find more information [here](https://stats.stackexchange.com/questions/1444/how-should-i-transform-non-negative-data-including-zeros) or [here](https://blogs.worldbank.org/impactevaluations/interpreting-treatment-effects-inverse-hyperbolic-sine-outcome-variable-and). 

By location and scale parameters, I just mean that the transformation is of the form:
```julia
x -> (asinh(loc + (x / scale / 2)) - loc) * scale
```
Which has 2 parameters, `loc` and `scale`, which need to be estimated. 

However, people will sometimes set `loc` to 0, giving a simplified transform of the form:
```julia
x -> asinh(x / scale / 2) * scale
```
Which only has one estimated parameter (`scale`).

**Comment by glevv** [comment]
This one is interesting, but will it be numerically stable?

**Comment by ParadaCarleton** [comment]
> This one is interesting, but is it numerically stable?

Yes, there shouldn't be any problems with it. The only possible numerical problem is if the data aren't scaled and mean-centered, you may have problems with fitting `loc` and `scale`. This should probably be mentioned in the docs.

**Comment by solegalli** [comment]
Hey guys! Thank you for the links and discussion. It looks good to me. Would you like to give it a go at drafting a class?

**Comment by ParadaCarleton** [comment]
@solegalli do you know how I can add a new transformer to the existing tests? I'm not sure where I can find the tests.

**Comment by solegalli** [comment]
You'd probably create a new .py with your transformer within the [transformation folder](https://github.com/feature-engine/feature_engine/tree/main/feature_engine/transformation). 

Then, you need to create another script within [this folder](https://github.com/feature-engine/feature_engine/tree/main/tests/test_transformation) where you'd add the tests.

Plus, you'd need to add your transformer to [this file](https://github.com/feature-engine/feature_engine/blob/main/tests/test_transformation/test_check_estimator_transformers.py) for generic tests, that may fail, but then i can help you troubleshoot.

**Comment by VascoSch92** [comment]
Hey,

can I implement this transformer? @solegalli 

is it still something to do?
